smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftSymbolProvider.kt

```diff
@@ -284,6 +297,88 @@ class SwiftSymbolProvider(private val model: Model, val swiftSettings: SwiftSett
+        // If default value is "null", it is explicit notation for no default value. Return unmodified builder.
+        if (defaultValueLiteral == "null") return builder
+
+        // The current shape may be a member shape or a root level shape.
+        val targetShape = when (shape) {
+            is MemberShape -> {
+                // If containing shape is an input shape, return unmodified builder.
+                if (model.expectShape(shape.container).hasTrait<InputTrait>()) return builder
+                model.expectShape(shape.target)
+            }
+            else -> shape
+        }
+
+        return when (targetShape) {
+            is ListShape -> builder.defaultValue("[]")
+            is EnumShape -> builder.defaultValue(".${swiftEnumCaseName(null, defaultValueLiteral)}")
+            is IntEnumShape -> {
+                // Get the corresponding enum member name (enum case name) for the int value from default trait
+                val enumMemberName = targetShape.enumValues.entries.firstOrNull {
+                    it.value == defaultValueLiteral.toInt()
+                }!!.key
+                builder.defaultValue(".${swiftEnumCaseName(enumMemberName, defaultValueLiteral)}")
+            }
+            is StringShape -> builder.defaultValue("\"$defaultValueLiteral\"")
+            is MapShape -> builder.defaultValue("[:]")
+            is BlobShape -> handleBlobDefaultValue(defaultValueLiteral, targetShape, builder)
+            is DocumentShape -> {
+                val node = shape.getTrait<DefaultTrait>()!!.toNode()
+                handleDocumentDefaultValue(defaultValueLiteral, node, builder)
+            }
+            is TimestampShape -> builder.defaultValue("Date(timeIntervalSince1970: $defaultValueLiteral)")
+            is FloatShape, is DoubleShape -> {
+                val decimal = ".0".takeIf { !defaultValueLiteral.contains(".") } ?: ""
+                builder.defaultValue(defaultValueLiteral + decimal)
+            }
+            // For: boolean, byte, short, integer, long, bigInteger, bigDecimal,
+            //      just take the literal string value from the trait.
+            else -> builder.defaultValue(defaultValueLiteral)
+        }
+    }
+
+    // Document: default value can be set to null, true, false, string, numbers, an empty list, or an empty map.
+    private fun handleDocumentDefaultValue(literal: String, node: Node, builder: Symbol.Builder): Symbol.Builder {
+        return when {
+            node.isObjectNode -> builder.defaultValue("Document.object([:])")
+            node.isArrayNode -> builder.defaultValue("Document.array([])")
+            node.isBooleanNode -> builder.defaultValue("Document.boolean($literal)")
+            node.isStringNode -> builder.defaultValue("Document.string(\"$literal\")")
+            node.isNumberNode -> builder.defaultValue("Document.number($literal)")
```

Comment:
[Smithy spec on document shape](https://smithy.io/2.0/spec/simple-types.html#document) says `arbitrary precision numbers` and given Swift's double range is e+308 and e-308 in magnitude in both positive and negative direction, I think it's safe to assume it will be a valid Swift number.